### PR TITLE
Settings context

### PR DESCRIPTION
**Pull request by pawtom**
A new module for using different Squish testSettings as context manager.

```    
with settings.logScreenshotOnFail(True), settings.logScreenshotOnPass(True):
        test.passes("aaa")   
        test.fail("bbb")
```

### CODE REVIEW

**Review by Adakar — CHANGES_REQUESTED** [review]
How about setting default values for parameters? At least for some of the functions.

File: squape/settings.py [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+@contextmanager
+def _ctx_settings(setting_name, value):
+
+    try:
+        current_value = getattr(squish.testSettings, setting_name)
+        setattr(squish.testSettings, setting_name, value)
+    except AttributeError as err:
+        raise AttributeError(
+            f"Your Squish version does not support test setting {setting_name}\n{err}"
+        )
```
Comment (Adakar) [inline squape/settings.py:21]: Could you create a dedicated Exception type that could be used by other modules in the future as well?

Comment (pawtom) [reply squape/settings.py]: Added custom SquishCapability exception

File: squape/settings.py [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+from contextlib import contextmanager
+
+import squish
+
+
+@contextmanager
+def _ctx_settings(setting_name, value):
+
```
[the remark this answers is not in the document]
Comment (pawtom) [reply squape/settings.py:23]: Docstring added

File: squape/internal/exceptions.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2023, Cyber Alpaca
+# All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+class SquishCapability(Exception):
```
Comment (Adakar) [inline squape/internal/exceptions.py:8]:
```suggestion
class SquishCapabilityError(Exception):
```

File: squape/settings.py [hunk 4]
```diff
@@ -29,85 +40,89 @@ def _ctx_settings(setting_name, value):
 
 
 @contextmanager
-def logScreenshotOnPass(enabled: bool) -> None:
+def logScreenshotOnPass(enabled=True) -> None:
```
Comment (Adakar) [inline squape/settings.py:43]:
```suggestion
def logScreenshotOnPass(enabled: bool = True) -> None:
```

File: squape/settings.py [hunk 5]
```diff
@@ -29,85 +40,89 @@ def _ctx_settings(setting_name, value):
...
     """
-    with _ctx_settings("logScreenshotOnPass", enabled) as c:
-        yield [c]
+    with _ctx_settings("logScreenshotOnPass", enabled):
+        yield
 
 
 @contextmanager
-def logScreenshotOnFail(enabled: bool) -> None:
+def logScreenshotOnFail(enabled=True) -> None:
```
Comment (Adakar) [inline squape/settings.py:60]:
```suggestion
def logScreenshotOnPass(enabled: bool = True) -> None:
```

File: squape/settings.py [hunk 6]
```diff
@@ -29,85 +40,89 @@ def _ctx_settings(setting_name, value):
...
 
-    with _ctx_settings("logScreenshotOnFail", enabled) as c:
-        yield [c]
+    with _ctx_settings("logScreenshotOnFail", enabled):
+        yield
 
 
 @contextmanager
-def logScreenshotOnWarning(enabled: bool) -> None:
+def logScreenshotOnWarning(enabled=True) -> None:
```
Comment (Adakar) [inline squape/settings.py:78]:
```suggestion
def logScreenshotOnWarning(enabled: bool = True) -> None:
```

File: squape/settings.py [hunk 7]
```diff
@@ -29,85 +40,89 @@ def _ctx_settings(setting_name, value):
...
-    squish.testSettings.logScreenshotOnError = enabled
-    yield
-    squish.testSettings.logScreenshotOnError = current_value
+    with _ctx_settings("logScreenshotOnWarning", enabled):
+        yield
 
 
 @contextmanager
-def silentVerifications(enabled: bool) -> None:
+def silentVerifications(enabled=True) -> None:
```
Comment (Adakar) [inline squape/settings.py:95]:
```suggestion
def silentVerifications(enabled: bool = True) -> None:
```

File: squape/settings.py [hunk 8]
```diff
@@ -29,85 +40,89 @@ def _ctx_settings(setting_name, value):
...
     """
-    with _ctx_settings("silentVerifications", enabled) as c:
-        yield [c]
+    with _ctx_settings("silentVerifications", enabled):
+        yield
 
 
 @contextmanager
-def imageSearchTolerant(enabled: bool) -> None:
+def imageSearchTolerant(enabled=True) -> None:
```
Comment (Adakar) [inline squape/settings.py:111]:
```suggestion
def imageSearchTolerant(enabled: bool = True) -> None:
```

File: squape/settings.py [hunk 9]
```diff
@@ -117,22 +132,31 @@ def imageSearchThreshold(threshold: float) -> None:
...
     """
-    with _ctx_settings("imageSearchThreshold", threshold) as c:
-        yield [c]
+    with _ctx_settings("imageSearchThreshold", threshold):
+        yield
 
 
 @contextmanager
-def imageSearchMultiscale(enabled: bool) -> None:
+def imageSearchMultiscale(enabled=True) -> None:
```
Comment (Adakar) [inline squape/settings.py:145]:
```suggestion
def imageSearchMultiscale(enabled: bool = True) -> None:
```

File: squape/settings.py [hunk 10]
```diff
@@ -218,23 +270,30 @@ def defaultOcrLanguage(language: str) -> None:
...
     """
-    with _ctx_settings("defaultOcrLanguage", language) as c:
-        yield [c]
+    with _ctx_settings("defaultOcrLanguage", language):
+        yield
 
 
 @contextmanager
-def breakOnFailure(enabled: bool) -> None:
+def breakOnFailure(enabled=True) -> None:
```
Comment (Adakar) [inline squape/settings.py:283]:
```suggestion
def breakOnFailure(enabled: bool = True) -> None:
```

**Review by Adakar — CHANGES_REQUESTED** [review]

File: squape/internal/exceptions.py (hunk 3 above)
Comment (pawtom) [reply squape/internal/exceptions.py]: The exception name was changed.

**Review by Adakar — APPROVED** [review]

**Pull request merged**